ihmc-perception/src/main/java/us/ihmc/perception/gpuHeightMap/FilteredRapidHeightMapExtractor.java

```diff
@@ -0,0 +1,162 @@
+package us.ihmc.perception.gpuHeightMap;
+
+import org.bytedeco.cuda.cudart.CUstream_st;
+import org.bytedeco.cuda.cudart.cudaExtent;
+import org.bytedeco.cuda.cudart.cudaPitchedPtr;
+import org.bytedeco.cuda.cudart.dim3;
+import org.bytedeco.javacpp.FloatPointer;
+import org.bytedeco.opencv.global.opencv_core;
+import org.bytedeco.opencv.opencv_core.GpuMat;
+import org.bytedeco.opencv.opencv_core.Mat;
+import us.ihmc.perception.cuda.CUDAKernel;
+import us.ihmc.perception.cuda.CUDAProgram;
+import us.ihmc.perception.cuda.CUDATools;
+
+import java.net.URL;
+
+import static org.bytedeco.cuda.global.cudart.*;
+
+public class FilteredRapidHeightMapExtractor
+{
+   static final int BLOCK_SIZE_XY = 32;
+
+   private final cudaPitchedPtr pointerTo3DArray;
+   private int currentIndex;
+   int layers;
```

Comment:
```suggestion
   private final int layers;
```

Also, sounds like this `layers` is the history length? Maybe that's a better name for the variable

Reply:
I left it as layers for now